File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int bgn=0,n=nums.size();
        for(int i=0;i<n;i++){
            if(nums[i]!=val){
                nums[bgn]=nums[i];
                bgn++;
            }
        }
        return bgn;
    }
};

File: 0027-remove-element/0027-remove-element.ts
function removeElement(nums: number[], val: number): number {
    const n = nums.length;
    let cnt = 0;
    for (let i = 0; i < n; i++) {
        if (val === nums[i]) {
            nums[i] = Infinity;
            cnt++
        }
    }
    nums.sort((a, b) => a - b);
    return n - cnt;
};